numpy_financial/_financial.py

```diff
@@ -767,28 +741,23 @@ def irr(values):
     if values.ndim != 1:
         raise ValueError("Cashflows must be a rank-1 array")
 
-    # Strip leading and trailing zeros. Since we only care about
-    # positive roots we can neglect roots at zero.
-    non_zero = np.nonzero(np.ravel(values))[0]
-    values = values[int(non_zero[0]):int(non_zero[-1])+1]
+    solution_found = False
+    p = np.polynomial.Polynomial(values)
+    pp = p.deriv()
```

Comment:
Could these names be more descriptive?

Reply:
```suggestion
    # NPV = V0 * x^0        + V1 * x^1         + ...
    #   let x = 1 / (1+eirr)
    # NPV = V0 * (1+eirr)^0 + V1 * (1+eirr)^-1 + ...
    # NPV = V0              + V1 * (1+eirr)^-1 + ...
    npv = np.polynomial.Polynomial(values)
    d_npv = p.deriv()
```

Reply:
Good idea. Do you think it would make more sense to swap lines 745 and 747/748?

Reply:
I've added some more description to the text around this as a separate suggestion (please excuse my OCD)